qemu/tests/ksmtuned.py

```diff
@@ -0,0 +1,88 @@
+import re
+import os
+import time
+import logging
+
+from shutil import copyfile
+
+from avocado.utils import process
+
+from virttest import env_process
+from virttest.utils_test import VMStress
+from virttest.staging import utils_memory
+
+
+def run(test, params, env):
+    """
+    Check KSM can be started automaticly when ksmtuned threshold is reached
+
+    1. Get the memory of your host and the KSM_THRES_COEF
+    2. Boot a guest with memory less than KSM_THRES_COEF threshold
+    3. Get the memory used in host of process qemu-kvm
+    4. Get the free memory in host
+    5. If both the free memory size is not smaller than the threshold and guest
+        used memory + threshold is not bigger than total memory in host. Check
+        the ksm status in host. Ksm should not start in the host
+    6. Repeat step 2~5 under it broke the rule in step 5
+
+    :param test: kvm test object.
+    :param params: Dictionary with test parameters.
+    :param env: Dictionary with the test environment.
+    """
+    def check_ksm(mem, stress=False):
+        """
+        :param mem: Boot guest with given memory
+        :param stress: Load stress or not
+        """
+        params['mem'] = mem // 1024
+        params['start_vm'] = 'yes'
+        vm_name = params['main_vm']
+        env_process.preprocess_vm(test, params, env, vm_name)
+        vm = env.get_vm(vm_name)
+        vm.verify_alive()
+        if stress:
+            params['stress_args'] = ('--cpu 4 --io 4 --vm 2 --vm-bytes %sM' %
+                                     (int(params['mem']) // 2))
+            stress_test = VMStress(vm, "stress", params)
+            stress_test.load_stress_tool()
+            time.sleep(10)
+        qemu_pid = vm.process.get_pid()
+        qemu_used_mem = int(process.getoutput(params['cmd_get_qemu_used_mem'] %
+                                              qemu_pid, shell=True)) * 4 * 1024
+        free_mem_host = utils_memory.freememtotal()
+        vm.destroy()
+        ksm_status = process.getoutput(params['cmd_check_ksm_status'])
```

Comment:
Let's get ksm status before vm.destroy.

Reply:
@huangyum  I don't know why but it can only passed with getting ksm status after vm.destroy.

Reply:
Extend the sleeping after loading stress to 30s, it passed, not sure if this long is proper.

Reply:
30s looks good to me.